custom-recipes/timeseries-preparation-resampling/recipe.py

```diff
@@ -39,7 +39,7 @@
     for column in schema
     if column["type"] in ["tinyint", "smallint", "int", "bigint"]
 ]
-output_df[columns_to_round] = output_df[columns_to_round].round()
+output_df[columns_to_round].astype(float).round().astype(int)
```

Comment:
i am wondering if this is maybe a bit slow/silly, and that we should only round if the column is already of a decimal type ? we're rounding to prevent decimal->integer errors, so in theory any column that is already an integer should be ok ?

Reply:
Yes, I think so too but funnily we are not rounding any decimal type now, are we ?

Reply:
![image](https://github.com/user-attachments/assets/a5bad1ad-032d-414a-bc61-4976b81ffab6)
I think we can target the pandas column correctly with the following "inexact" targeting coming from [numpys dtypes](https://numpy.org/doc/stable/reference/arrays.scalars.html) hierarchy.

Reply:
Therefore [8eee3c5](https://github.com/dataiku/dss-plugin-timeseries-preparation/pull/62/commits/8eee3c5e01ff2257a9e100938b56fee9ea3ee880). Let me know what you think @tmwly